raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -77,10 +77,17 @@ def schedule_task(
         if callback is not None:
             assert callable(callback), "must be a callable"
 
-            def task_done(result: asyncio.Future) -> None:
-                # spawn a new greenlet that invokes the callback with the
-                # coroutine's result and provided args/kwargs
-                self.schedule_greenlet(callback, result.result(), *args, **kwargs)
+            def task_done(future: asyncio.Future) -> None:
+                try:
+                    result = future.result()
+                except asyncio.CancelledError:
+                    pass
+                except Exception:  # pylint: disable=broad-except
+                    log.exception("Exception raised by task %r", task)
```

Comment:
Unless I'm misunderstanding, this doesn't look correct. 
It will ignore task exceptions (apart from logging them).

Reply:
Won't the exception actually be caught and dealt with by whoever is really awaiting on the task?